### PR TITLE
Restrict ffmpeg to 4.2+.X versions to resolve linux conda build failures

### PR DESCRIPTION
**Pull request by atalman**
Looks like ffmpeg release 6.1.1 broke our linux conda builds: https://anaconda.org/anaconda/ffmpeg/files?version=6.1.1
Fixes: https://github.com/pytorch/vision/issues/8560

### CODE REVIEW

File: .github/workflows/build-conda-linux.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
             smoke-test-script: test/smoke_test.py
             package-name: torchvision
     name: ${{ matrix.repository }}
-    uses: pytorch/test-infra/.github/workflows/build_conda_linux.yml@main
+    uses: pytorch/test-infra/.github/workflows/build_conda_linux.yml@atalman-conda-build
```
Comment (atalman) [inline .github/workflows/build-conda-linux.yml:37]: Will change before pushing

File: packaging/torchvision/meta.yaml [hunk 2]
```diff
@@ -26,11 +26,11 @@ requirements:
     - numpy >=1.23.5 # [py >= 311]
     - requests
     - libpng
-    - ffmpeg >=4.2  # [linux]
+    - ffmpeg ==4.2.2  # [linux]
```
Comment (malfet) [inline packaging/torchvision/meta.yaml:29]:
```suggestion
    - ffmpeg >=4.2.2, <5.0.0  # [linux]
```

**Review by malfet — COMMENTED** [review]
What's wrong with ffmpeg-4.3? (I.e. why pin it to 4.4.2, rather than allow the range?)

File: packaging/torchvision/meta.yaml [hunk 3]
```diff
@@ -11,13 +11,13 @@ requirements:
     - {{ compiler('c') }} # [win]
     - libpng
     - libjpeg-turbo
-    - ffmpeg >=4.2  # [linux]
+    - ffmpeg ==4.2.2  # [linux]
```
Comment (malfet) [inline packaging/torchvision/meta.yaml:14]:
```suggestion
    - ffmpeg >=4.2.2, <5.0.0  # [linux]
```

**Review by malfet — APPROVED** [review]

**Comment by atalman** [comment]
Anaconda has 4.2.2 and 6.1.1 but I think its ok to pin it to range
Screenshot 2024-07-30 at 6 17 56 PM

**Pull request merged**